### PR TITLE
[MAINTENANCE] Merge scheduler-plugin into master

### PR DESCRIPTION
**Pull request by e-cuellar**

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 3h 26m 30s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 2 test(s) failed:

<details><summary><b>org.pentaho.platform.plugin.services.exporter.PentahoPlatformExporterTest.testExportSchedules_SchedulereThrowsException</b> (click to expand)</summary><p>

```

Wanted but not invoked:
iScheduler.getJobs(null);
-> at org.pentaho.platform.plugin.services.exporter.PentahoPlatformExporterTest.testExportSchedules_SchedulereThrowsException(PentahoPlatformExporterTest.java:154)
Actually, there were zero interactions with this mock.

```

</p></details>

<details><summary><b>org.pentaho.platform.plugin.services.importer.SolutionImportHandlerTest.testFileIsScheduleInputSource</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

  
Tests run: **2676**, Failures: **2**, Skipped: **6** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5493/1/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: user-console/src/main/java/org/pentaho/mantle/client/commands/RunInBackgroundCommand.java [hunk 1]
```diff
@@ -459,26 +442,55 @@ public void onResponseReceived( Request request, Response response ) {
...
+  private native void createScheduleParamsDialog( String filePath, JSONObject scheduleRequest, Boolean isEmailConfigValid, Boolean isSchedulesPerspectiveActive ) /*-{
+   $wnd.pho.createScheduleParamsDialog( filePath, scheduleRequest, isEmailConfigValid, isSchedulesPerspectiveActive );
+  }-*/;
+
+  private native void createScheduleEmailDialog( String filePath, JSONObject scheduleRequest ) /*-{
+   $wnd.pho.createScheduleEmailDialog( filePath, scheduleRequest );
+  }-*/;
+
+  private native void getScheduleParams( JSONObject scheduleRequest) /*-{
+   $wnd.pho.getScheduleParams( scheduleRequest );
```
Comment (rmansoor) [inline user-console/src/main/java/org/pentaho/mantle/client/commands/RunInBackgroundCommand.java:494]: @e-cuellar  Nick has found an issue with this code. Please review this slack history
https://hitachivantara-eng.slack.com/archives/C01HJ874L8Z/p1703281047403969

File: assemblies/pentaho-server/pom.xml [hunk 2]
```diff
@@ -55,6 +55,7 @@
       <groupId>org.hsqldb</groupId>
       <artifactId>hsqldb</artifactId>
       <version>${hsqldb.version}</version>
+<!--      <classifier>jdk8</classifier>-->
```
Comment (rmansoor) [inline assemblies/pentaho-server/pom.xml:58]: @wseyler  do you know why this change was made

File: assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/pentahoServices.spring.xml [hunk 3]
```diff
@@ -134,8 +128,8 @@
   </bean>
   <bean class="org.pentaho.platform.web.http.api.resources.EmailResource" scope="request"/>
   <bean class="org.pentaho.platform.web.http.api.resources.SessionResource" scope="request"/>
-  <bean class="org.pentaho.platform.web.http.api.resources.SchedulerResource" scope="request"/>
```
Comment (rmansoor) [inline assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/pentahoServices.spring.xml:137]: We need to remove this code not comment it

File: user-console/src/main/java/org/pentaho/mantle/client/commands/RunInBackgroundCommand.java (hunk 1 above)
Comment (e-cuellar) [reply user-console/src/main/java/org/pentaho/mantle/client/commands/RunInBackgroundCommand.java]: Implemented Nic's changes @rmansoor

File: assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/pentahoServices.spring.xml (hunk 3 above)
Comment (e-cuellar) [reply assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/pentahoServices.spring.xml:137]: Rmoved this commented code @rmansoor

**Comment by buildguy** [comment]
### :x: Build failed in 3h 16m
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 2 test(s) failed:

<details><summary><b>org.pentaho.platform.plugin.services.exporter.PentahoPlatformExporterTest.testExportSchedules_SchedulereThrowsException</b> (click to expand)</summary><p>

```

Wanted but not invoked:
iScheduler.getJobs(null);
-> at org.pentaho.platform.plugin.services.exporter.PentahoPlatformExporterTest.testExportSchedules_SchedulereThrowsException(PentahoPlatformExporterTest.java:154)
Actually, there were zero interactions with this mock.

```

</p></details>

<details><summary><b>org.pentaho.platform.plugin.services.importer.SolutionImportHandlerTest.testFileIsScheduleInputSource</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

  
Tests run: **2676**, Failures: **2**, Skipped: **6** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5493/2/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 3h 17m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 2 test(s) failed:

<details><summary><b>org.pentaho.platform.plugin.services.exporter.PentahoPlatformExporterTest.testExportSchedules_SchedulereThrowsException</b> (click to expand)</summary><p>

```

Wanted but not invoked:
iScheduler.getJobs(null);
-> at org.pentaho.platform.plugin.services.exporter.PentahoPlatformExporterTest.testExportSchedules_SchedulereThrowsException(PentahoPlatformExporterTest.java:154)
Actually, there were zero interactions with this mock.

```

</p></details>

<details><summary><b>org.pentaho.platform.plugin.services.importer.SolutionImportHandlerTest.testFileIsScheduleInputSource</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

  
Tests run: **2676**, Failures: **2**, Skipped: **6** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5493/3/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_
